libs/leonsans/src/leonsans.ts

```diff
@@ -0,0 +1,608 @@
+  position(x = 0, y = 0) {
+    if (this.model.position(x, y)) {
+      this.updateId_++;
+      this.updateSignal();
+    }
+  }
+
+  update() {
+    this.updateId_++;
+
+    this.model.update(
+      this.str_,
+      this.width_,
+      this.breakWord_,
+      this.weight_,
+      this.size_,
+      this.tracking_,
+      this.leading_,
+    );
+
+    if (this.isPattern_ || this.isWave_) {
+      this.updatePatternPaths();
+      this.updateWavePaths();
+    } else {
+      this.updateSignal();
+    }
+  }
+
+  updateGuide() {
+    if (this.guideId_ != this.updateId_) {
+      this.guideId_ = this.updateId_;
+      this.model.updateGuide();
+    }
+  }
+
+  /**
+   * Update paths for drawing in WebGL (PIXI.js). It's very expensive, only call when it needs.
+   */
+  updateDrawingPaths() {
+    if (this.drawingPathsId_ != this.updateId_) {
+      this.drawingPathsId_ = this.updateId_;
+      this.model.updateDrawingPaths();
+    }
+  }
+
+  /**
+   * Update paths for pattern
+   * @param {boolean} force - Force execution
+   */
+  updatePatternPaths(force?: boolean) {
+    if (this.isPattern_ && (force || this.patternPathsId_ != this.updateId_)) {
+      this.patternPathsId_ = this.updateId_;
+      this.model.updatePatternPaths(this.pathGap_);
+      this.isForceRender_ = true;
+      this.updateSignal();
+    }
+  }
+
+  /**
+   * Update paths for wave effect
+   * @param {boolean} force - Force execution
+   */
+  updateWavePaths(force?: boolean) {
+    if (this.isWave_ && (force || this.wavePathsId_ != this.updateId_)) {
+      this.wavePathsId_ = this.updateId_;
+      this.model.updateWavePaths(this.pathGap_);
+      this.isForceRender_ = true;
+      this.updateSignal();
+    }
+  }
+
+  updateSignal() {
+    this.model.updateLinesForRect();
+    this.model.updatePathsForRect();
+    this.dispatch('update', this.model);
+  }
+
+  reset() {
+    this.size_ = 500;
+    this.weight_ = MIN_FONT_WEIGHT;
+    this.color_ = ['#000000'];
+    this.tracking_ = 0;
+    this.leading_ = 0;
+    this.pathGap_ = 0.5;
+    this.amplitude_ = 0.5;
+    this.width_ = 0;
+    this.breakWord_ = false;
+    this.fps_ = 30;
+    this.frameInterval_ = 1000 / this.fps_;
+    this.isPattern_ = false;
+    this.isWave_ = false;
+    this.hasGrid_ = false;
+
+    this.str_ = '';
+
+    this.time_ = 0;
+    this.isOutdated_ = false;
+    this.isForceRender_ = false;
+
+    this.updateId_ = 0;
+    this.drawingPathsId_ = 0;
+    this.patternPathsId_ = 0;
+    this.wavePathsId_ = 0;
+    this.guideId_ = 0;
+
+    this.model.reset();
+  }
+
+  dispose() {
+    this.reset();
+    this.model = new Model();
+  }
+
+  /**
+   * Draw text in WebGL with PIXI.js
+   * @param {PIXI.Graphics} graphics
+   */
+  drawPixi(graphics: Graphics) {
+    const total = this.model.data.length;
+    let i, d, color;
+    for (i = 0; i < total; i++) {
+      d = this.model.data[i];
+      color = PixiColor(i, d, this.color_);
+      PixiLines(graphics, d, this.lineWidth, color);
+    }
```

Comment:
```suggestion
    this.model.data.forEach((datum, index) => {
      const color = PixiColor(index, datum, this.color_);
      PixiLines(graphics, datum, this.lineWidth, color);
    });
```

forEach를 안 쓰신 건 성능 때문인가요?? 뭐 바꾸자는 건 아니고 그냥 궁금해스